File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (i1, i2) -> i1[0] - i2[0]);
        List<List<Integer>> mergedIntervals = new ArrayList<>();

        int start = intervals[0][0];
        int end = intervals[0][1];
        for (int[] interval : intervals) {
            if (interval[0] <= end) {
                start = Math.min(start, interval[0]);
                end = Math.max(end, interval[1]);
            } else {
                ArrayList<Integer> overlappedInterval = new ArrayList<>();
                overlappedInterval.add(start);
                overlappedInterval.add(end);
                mergedIntervals.add(overlappedInterval);
                start = interval[0];
                end = interval[1];
            }
        }

        ArrayList<Integer> overlappedInterval = new ArrayList<>();
        overlappedInterval.add(start);
        overlappedInterval.add(end);
        mergedIntervals.add(overlappedInterval);

        return mergedIntervals.stream()
                .map(interval -> interval.stream().mapToInt(i -> i).toArray())
                .toArray(int[][]::new);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a, b) => a[0] - b[0]);
    const mergedIntervals = [];
    let intervalStart = intervals[0][0];
    let intervalEnd = intervals[0][1];

    for (let i = 1; i < intervals.length; i++) {
        if (intervalEnd >= intervals[i][0]) {
            intervalEnd = intervals[i][1];
        } else {
            mergedIntervals.push([intervalStart, intervalEnd]);
            intervalStart = intervals[i][0];
            intervalEnd = intervals[i][1];
        }
    }
    mergedIntervals.push([intervalStart, intervalEnd]);
    return mergedIntervals;
};